Detections/ASimAuthentication/imAuthSigninsMultipleCountries.yaml

```diff
@@ -23,20 +23,28 @@ query: |
   let threshold = 2;
   imAuthentication
   | where TimeGenerated > timeframe
-  | where EventType=='Logon' and EventResult=='Success'
+  | where EventType == 'Logon' and EventResult == 'Success'
   | where isnotempty(SrcGeoCountry)
-  | summarize StartTime = min(TimeGenerated), EndTime = max(TimeGenerated), Vendors=make_set(EventVendor), Products=make_set(EventProduct)
-    , NumOfCountries = dcount(SrcGeoCountry)
-    by TargetUserId, TargetUsername, TargetUserType
+  | summarize
+      StartTime        = min(TimeGenerated)
+      , EndTime        = max(TimeGenerated)
+      , Vendors        = make_set(EventVendor, 128)
+      , Products       = make_set(EventProduct, 128)
+      , NumOfCountries = dcount(SrcGeoCountry)
+      , Countries      = make_set(SrcGeoCountry, 128)
+      by TargetUserId, TargetUsername, TargetUserType
   | where NumOfCountries >= threshold
   | extend timestamp = StartTime, AccountCustomEntity = TargetUsername
-
 entityMappings:
   - entityType: Account
     fieldMappings:
-      - identifier: FullName
-        columnName: AccountCustomEntity
-version: 1.2.2
+      - identifier: Name
+        columnName: Account
+  - entityType: Account
+    fieldMappings:
+      - identifier: UPNSuffix
```

Comment:
Entity is not required to be repeated. Please include the UPN identifier in the same Entity as the Name.
Also, you would have to split the value of TargetUsername on "@" to get the Name and UPNSuffix.

Reply:
I just thought to simply pick up an issue of an existing rule and add the requested field. Didn't expect so many other changes o  a rule that is already in production.

Will add the requested changes later tonight